Add page object patterns

smerdoff opened the pull request:
- Added CheckoutPage
- Added 5 tests in the SauceDemoTest class
- Added Pages from lecture

smerdoff committed:
-Added CheckoutPage
-Added 5 tests
- Added Pages from lecture

smerdoff committed:
Edited

dzmitryrak commented on src/test/java/pages/CheckoutPage.java:
```diff
@@ -0,0 +1,52 @@
+
+    public void zipCodePlaceholderTest(){
+        String zipCodePlaceholder = driver.findElement(ZIP_INPUT).getAttribute("placeholder");
+        Assert.assertEquals(zipCodePlaceholder, expectedZipCodePlaceholder, "Не совпадают плейсхолдер имени");
```
копипаст сообщения

dzmitryrak commented on src/test/java/pages/LoginPage.java:
```diff
@@ -0,0 +1,48 @@
+        driver.findElement(PASSWORD_INPUT).sendKeys(password);
+        driver.findElement(LOGIN_BUTTON).click();
+        String actualText =  driver.findElement(ERROR_TEXT).getText();
+        Assert.assertEquals(actualText,expectedErrorOfLockedUser, "Текст не совпадает" );
```
тоже сообщения не ок

dzmitryrak commented on src/test/java/tests/SauceDemoTest.java:
```diff
@@ -0,0 +1,68 @@
+        productsPage.addToCart("Sauce Labs Onesie");
+        cartPage.openPage();
+        cartPage.clickCheckout();
+        checkoutPage.fillInputs("Firstname", "LastName", "11111");
```
Нет валидаций дальнейших после чекаута. Добавить

dzmitryrak commented:
Потом еще раз мне скинь

smerdoff committed:
-Изменил тесты строго согласно комментариям
- Добавил ещё один тест на валидацию Checkout Overview страницы
- +изменения в проекте после лекции

smerdoff commented on src/test/java/tests/SauceDemoTest.java:
```diff
@@ -37,6 +38,23 @@ public void checkoutFields(){
         cartPage.openPage();
         cartPage.clickCheckout();
         checkoutPage.fillInputs("Firstname", "LastName", "11111");
+        checkoutPage.checkoutStepTwoURLValidation();
```
Валидация после чекаута

dzmitryrak commented:
Вмержил. Отвечай плиз на комменты в PR когда фиксишь
